CHANGELOG.md

```diff
@@ -1,3 +1,17 @@
+## 4.0.0 
+* Using version 2 of the notification-api.
+* Update to `NotificationsAPIClient.send_sms_notification()`: 
+    * added `reference`: an optional unique identifier for the notification or an identifier for a batch of notifications. `reference` can be omitted.
+    * the `dict` returned has changed, learn more from the README. 
+* Update to `NotificationClient.send_email_notification()`: 
+    * added `reference`: an optional unique identifier for the notification or an identifier for a batch of notifications. `reference` can be omitted.
+    * the `dict` returned has changed, learn more from the README. 
+* Update to the response of `NotificationsAPIClient.get_notification_by_id(id)`, see the README for details.
+* `NotificationsAPIClient.get_all_notifications()` 
+    * Notifications can be now be filtered by `reference`, see the README for details.
+    * NotificationList response has changed, see the README for details.
```

Comment:
should we remove `service_id` from  BaseAPIClient (and mention it here?) like we've done in the other languages?